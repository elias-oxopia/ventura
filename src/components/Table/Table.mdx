---
name: Table
menu: Components
route: /table
---

import { Playground, Props } from 'docz';
import { Table } from '../..';

# Table

A table displays data in a familiar structure, making it easy to read.

## Best practices

Use the different building block to correctly structure your table:

- `Table`
- `Table.Header`
- `Table.HeaderCell`
- `Table.Body`
- `Table.Row`
- `Table.Cell`

## Examples

### Basic usage

<Playground>
    <Table>
      <Table.Header>
        <Table.HeaderCell>
          First name
        </Table.HeaderCell>
        <Table.HeaderCell>
          Last name
        </Table.HeaderCell>
      </Table.Header>
      <Table.Body>
        <Table.Row>
          <Table.Cell>
            Bram
          </Table.Cell>
          <Table.Cell>
            Vanhoutte
          </Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell>
            Robin
          </Table.Cell>
          <Table.Cell>
            Wijnant
          </Table.Cell>
        </Table.Row>
      </Table.Body>
    </Table>
</Playground>

## API

### Table

<Props of={Table} />

### Table.Header

<Props of={Table.Header} />

### Table.HeaderCell

<Props of={Table.HeaderCell} />

### Table.Body

<Props of={Table.Body} />

### Table.Row

<Props of={Table.Row} />

### Table.Row

<Props of={Table.Cell} />
