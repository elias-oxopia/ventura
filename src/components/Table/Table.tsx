import React from 'react';
import classNames from 'classnames';

import Body from './TableBody/TableBody';
import Cell from './TableCell/TableCell';
import Header from './TableHeader/TableHeader';
import HeaderCell from './TableHeaderCell/TableHeaderCell';
import Row from './TableRow/TableRow';

import cssReset from '../../css-reset.module.css';
import styles from './Table.module.css';

type Props = {
  className?: string;
  children?: React.ReactNode;
};

const Table: React.FC<Props> & {
  Header: typeof Header;
  HeaderCell: typeof HeaderCell;
  Body: typeof Body;
  Row: typeof Row;
  Cell: typeof HeaderCell;
} = ({ className, children }: Props) => {
  const tableClassNames = classNames(cssReset.ventura, styles.table, className);

  return <table className={tableClassNames}>{children}</table>;
};

Table.Header = Header;
Table.HeaderCell = HeaderCell;
Table.Body = Body;
Table.Row = Row;
Table.Cell = Cell;

export default Table;
