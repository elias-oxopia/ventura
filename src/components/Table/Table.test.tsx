import React from 'react';
import { render } from '@testing-library/react';

import Table from './Table';

interface Course {
  name: string;
  length: string;
  status: 'done' | 'in progress' | 'not started';
}

describe('Table', () => {
  const courses: Course[] = [
    { name: 'Course 1', length: '2h', status: 'done' },
    { name: 'Course 2', length: '4h', status: 'in progress' },
    { name: 'Course 3', length: '1h', status: 'in progress' },
    { name: 'Course 4', length: '3h 43m', status: 'not started' },
    { name: 'Course 5', length: '1h', status: 'in progress' },
    { name: 'Course 6', length: '2h 15m', status: 'not started' },
    { name: 'Course 7', length: '3h', status: 'done' },
  ];

  test('default snapshot', () => {
    const component = (
      <Table>
        <Table.Header>
          <Table.HeaderCell>Course</Table.HeaderCell>
          <Table.HeaderCell>Length</Table.HeaderCell>
          <Table.HeaderCell>Status</Table.HeaderCell>
        </Table.Header>
        <Table.Body>
          {courses.map((course) => {
            return (
              <Table.Row>
                <Table.Cell>{course.name}</Table.Cell>
                <Table.Cell>{course.length}</Table.Cell>
                <Table.Cell>{course.status}</Table.Cell>
              </Table.Row>
            );
          })}
        </Table.Body>
      </Table>
    );
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });
});
