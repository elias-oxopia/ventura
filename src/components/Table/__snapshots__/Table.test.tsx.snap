// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table default snapshot 1`] = `
<DocumentFragment>
  <table
    class="ventura table"
  >
    <thead>
      <tr>
        <th
          class="cellAlignLeft headerCell"
        >
          Course
        </th>
        <th
          class="cellAlignLeft headerCell"
        >
          Length
        </th>
        <th
          class="cellAlignLeft headerCell"
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="row"
      >
        <td
          class="cellAlignLeft cell"
        >
          Course 1
        </td>
        <td
          class="cellAlignLeft cell"
        >
          2h
        </td>
        <td
          class="cellAlignLeft cell"
        >
          done
        </td>
      </tr>
      <tr
        class="row"
      >
        <td
          class="cellAlignLeft cell"
        >
          Course 2
        </td>
        <td
          class="cellAlignLeft cell"
        >
          4h
        </td>
        <td
          class="cellAlignLeft cell"
        >
          in progress
        </td>
      </tr>
      <tr
        class="row"
      >
        <td
          class="cellAlignLeft cell"
        >
          Course 3
        </td>
        <td
          class="cellAlignLeft cell"
        >
          1h
        </td>
        <td
          class="cellAlignLeft cell"
        >
          in progress
        </td>
      </tr>
      <tr
        class="row"
      >
        <td
          class="cellAlignLeft cell"
        >
          Course 4
        </td>
        <td
          class="cellAlignLeft cell"
        >
          3h 43m
        </td>
        <td
          class="cellAlignLeft cell"
        >
          not started
        </td>
      </tr>
      <tr
        class="row"
      >
        <td
          class="cellAlignLeft cell"
        >
          Course 5
        </td>
        <td
          class="cellAlignLeft cell"
        >
          1h
        </td>
        <td
          class="cellAlignLeft cell"
        >
          in progress
        </td>
      </tr>
      <tr
        class="row"
      >
        <td
          class="cellAlignLeft cell"
        >
          Course 6
        </td>
        <td
          class="cellAlignLeft cell"
        >
          2h 15m
        </td>
        <td
          class="cellAlignLeft cell"
        >
          not started
        </td>
      </tr>
      <tr
        class="row"
      >
        <td
          class="cellAlignLeft cell"
        >
          Course 7
        </td>
        <td
          class="cellAlignLeft cell"
        >
          3h
        </td>
        <td
          class="cellAlignLeft cell"
        >
          done
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
